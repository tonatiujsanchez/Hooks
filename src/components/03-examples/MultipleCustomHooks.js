import useCounter from "../../hooks/useCounter"
import useFetch from "../../hooks/useFetch"


import Header from "./Header"
// import Quote from "./Quote"
import Loading from "./Loading"


const MultipleCustomHooks = () => {

    const { counter, increment, decrement } = useCounter(1)
    const { data, loading } = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`)

    const { author, quote } = !!data && data[0]
    // const {author, quote} = data ? data[0]: {}


    return (

        <div className="row">
            <div className="col-md-10 col-lg-8 mx-auto">
                <Header increment={increment} decrement={decrement} counter={counter} />
                <hr />
                {loading
                    ? <Loading />
                    : <blockquote className="blockquote">
                        <p className="mt-5 mb-4 text-center">{quote}</p>
                        <figcaption className="blockquote-footer text-center">
                            <cite title="Source Title">{author}</cite>
                        </figcaption>
                    </blockquote>
                }
            </div>

        </div>
    )
}

export default MultipleCustomHooks