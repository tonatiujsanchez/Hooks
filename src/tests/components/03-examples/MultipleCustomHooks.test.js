import { shallow } from "enzyme"
import MultipleCustomHooks from "../../../components/03-examples/MultipleCustomHooks"
import useFetch from "../../../hooks/useFetch"
import useCounter from "../../../hooks/useCounter"

jest.mock('../../../hooks/useFetch')
jest.mock('../../../hooks/useCounter')


describe('Pruebas en <MultipleCustomHooks />', () => { 

    beforeEach( () => {
        useFetch.mockReturnValue({ data: null, loading: true, error: null })
    })
    
    beforeEach( () => {
        useCounter.mockReturnValue({ 
            counter: 1,
            increment: ()=>{},
            decrement: ()=>{},
         })
    })

    
    test('Debe de mostrarse correctamente', () => { 

        const wrapper = shallow( <MultipleCustomHooks /> )
        expect( wrapper ).toMatchSnapshot()
     })


     test('Debe de mostrar la informaciÃ³n', () => { 

        useFetch.mockReturnValue({ 
            data: [{
                author: 'Brandon',
                quote: 'Hello world!'
            }], 
            loading: false, 
            error: null 
        })

        const wrapper = shallow( <MultipleCustomHooks /> )


        const loading = wrapper.find('Loading')
        expect( loading.exists() ).toBe( false )
        
        const quote = wrapper.find('.blockquote p').text().trim()
        expect( quote).toBe( 'Hello world!' )

        const author = wrapper.find('cite').text().trim()
        expect( author ).toBe('Brandon')
        

      })


 })