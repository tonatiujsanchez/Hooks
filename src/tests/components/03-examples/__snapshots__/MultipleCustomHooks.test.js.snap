// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks /> Debe de mostrarse correctamente 1`] = `
<div
  className="row"
>
  <div
    className="col-md-10 col-lg-8 mx-auto"
  >
    <Header
      counter={1}
      decrement={[Function]}
      increment={[Function]}
    />
    <hr />
    <Loading />
  </div>
</div>
`;
